src/PaymentSource.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Invoiced;
+
+class PaymentSource extends BaseObject
+{
+    use Operations\All;
+
+    /**
+     * Fetches a collection of objects. This variant constructs them into Card or BankAccount objects as appropriate.
+     *
+     * @param array $opts
+     *
+     * @return [array(Invoiced\Object), Invoiced\Collection]
+     */
+    public function all(array $opts = [])
+    {
+        $response = $this->_client->request('get', $this->getEndpoint(), $opts);
+
+        $output = [];
+
+        // create card and bank account objects appropriately
+        foreach ($response['body'] as $obj) {
+            if ($obj['object'] == 'card') {
+                $card = new Card($this->getClient());
+                $card->setEndpointBase($this->getEndpointBase());
+                array_push($output, Util::convertToObject($card, $obj));
```

Comment:
It's better to use `$output[] = ...` instead of `array_push()`

Reply:
got it, making this change